### PR TITLE
:memo: Add inpaint examples

### LINKED ISSUE #2310: [Bug]: Inpaint via API txt2img not working
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

I am attempting to use txt2img and controlnet with an image and a mask, but I'm encountering issues where the mask seems ineffective. This is a shift from my previous workflow, where I used im2img without controlnet for inpainting. Now, my goal is to utilize txt2img with controlnet for the same purpose. Despite reviewing both resolved and open issues, and examining the payload through an extension, I am unable to diagnose the problem.

**PLEASE HELP**, **I've been stuck on this for the last 2 months now** 

In the example below, I'm trying to close her eyes via API without success, I've already tried a thousand things;

- img2img controlnet
- txt2img controlnet
- input_image and mask separately
- input_image: {input_image: base64string, mask: base64mask}
- input_image: {image: base64string, mask: base64mask}
- image: [base64string, base64mask]
- image: {image: base64string, mask: base64mask}
- mask white/black and black/white

Payload used in this example:
{        
        "prompt": "(grainy cinematic photography:1.2) shot on Bessa R2A Cinestill photo of a beautiful young italian woman, wavy hair, cute smile, closed eyes",
        "negative_prompt": "",
        "sampler_name": "DPM++ 2M Karras",
        "batch_size": 1,
        "steps": 20,
        "cfg_scale": 7,
        "width": 512,
        "height": 768,
        "alwayson_scripts": {
            "controlnet": {
                "args": [
                    {
                        "model": "control_v11p_sd15_inpaint [ebff9138]",
                        "module": "inpaint_only+lama",
                        "resize_mode": 1,
                        "control_mode": 0,
                        "image": {"image": base64string, "mask": base64mask}
                    }
                ]
            }
        }
}

Image:
original

Mask:
inpaint_mask

Results via UI:
a1111_inpainted_txt2imgg

Results via API is the exact same original image.
download

Mask returned from the API (second image from the response)
download

### Steps to reproduce the problem

1. Go to http://127.0.0.1:7860/docs#/default/text2imgapi_sdapi_v1_txt2img_post
2. Use the payload provided
3. Help

### What should have happened?

The response from the API being the same as the UI.

### Commit where the problem happens

webui: txt2img
controlnet: inpaint

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
--opt-sdp-attention --api --share --enable-insecure-extension-access
```

### List of enabled extensions

Controlnet, Dynamic Prompts, Additional Networks, Adetailer, AnimateDiff

### Console logs

```Shell
venv "F:\AI\AUTOMATIC1115\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.6.1
Commit hash: 4afaaf8a020c1df457bcf7250cb1c7f609699fa7
Launching Web UI with arguments: --opt-sdp-attention --api --share --enable-insecure-extension-access
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
[-] ADetailer initialized. version: 23.11.0, num models: 11
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
2023-12-08 17:46:46,003 - ControlNet - INFO - ControlNet v1.1.417
ControlNet preprocessor location: F:\AI\AUTOMATIC1115\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2023-12-08 17:46:46,112 - ControlNet - INFO - ControlNet v1.1.417
Loading weights [ce4629b477] from F:\AI\AUTOMATIC1115\stable-diffusion-webui\models\Stable-diffusion\Real_Jugg.safetensors
2023-12-08 17:46:47,272 - AnimateDiff - INFO - Injecting LCM to UI.
2023-12-08 17:46:47,569 - AnimateDiff - INFO - Hacking i2i-batch.
Creating model from config: F:\AI\AUTOMATIC1115\stable-diffusion-webui\models\Stable-diffusion\Real_Jugg.yaml
Applying attention optimization: sdp... done.
Model loaded in 6.8s (load weights from disk: 0.6s, create model: 0.7s, apply weights to model: 3.3s, apply half(): 1.1s, calculate empty prompt: 1.0s).
F:\AI\AUTOMATIC1115\stable-diffusion-webui\extensions\sd-webui-additional-networks\scripts\metadata_editor.py:399: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  with gr.Row().style(equal_height=False):
F:\AI\AUTOMATIC1115\stable-diffusion-webui\extensions\sd-webui-additional-networks\scripts\metadata_editor.py:521: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  cover_image = gr.Image(
Running on local URL:  http://127.0.0.1:7860
Running on public URL: https://524f7e3577b2b61905.gradio.live

This share link expires in 72 hours. For free permanent hosting and GPU upgrades, run `gradio deploy` from Terminal to deploy to Spaces (https://huggingface.co/spaces)
Startup time: 27.0s (prepare environment: 2.7s, import torch: 2.8s, import gradio: 0.7s, setup paths: 0.6s, initialize shared: 0.2s, other imports: 0.5s, list SD models: 0.2s, load scripts: 3.6s, create ui: 7.2s, gradio launch: 8.4s).
2023-12-08 17:48:27,418 - ControlNet - INFO - Loading model: control_v11p_sd15_inpaint [ebff9138]
2023-12-08 17:48:27,864 - ControlNet - INFO - Loaded state_dict from [F:\AI\AUTOMATIC1115\stable-diffusion-webui\extensions\sd-webui-controlnet\models\control_v11p_sd15_inpaint.pth]
2023-12-08 17:48:27,864 - ControlNet - INFO - controlnet_default_config
2023-12-08 17:48:30,114 - ControlNet - INFO - ControlNet model control_v11p_sd15_inpaint [ebff9138] loaded.
2023-12-08 17:48:30,212 - ControlNet - INFO - using inpaint as input
2023-12-08 17:48:30,212 - ControlNet - INFO - Loading preprocessor: inpaint_only+lama
2023-12-08 17:48:30,212 - ControlNet - INFO - preprocessor resolution = -1
2023-12-08 17:48:31,562 - ControlNet - INFO - ControlNet used torch.float16 VAE to encode torch.Size([1, 4, 96, 64]).
2023-12-08 17:48:31,579 - ControlNet - INFO - ControlNet Hooked - Time = 4.364281177520752
2023-12-08 17:48:31,629 - ControlNet - INFO - [ControlNet] Initial noise hack applied to torch.Size([1, 4, 96, 64]).
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:06<00:00,  3.09it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:05<00:00,  3.38it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:05<00:00,  4.27it/s]
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by huchenlei**
Closes #2310.

This PR adds API example for both txt2img inpaint and img2img inpaint.

txt2img inpaint result:
txt2img-0
txt2img-1

img2img inpaint result:
img2img-0
img2img-1

Model used: cetusMix_v4.safetensors [b42b09ff12]

### CODE REVIEW

**Comment by huchenlei** [comment]
Closes 2242.

**Comment by AIGC404** [comment]
What is the SD version you are running?

**Comment by AIGC404** [comment]
I ran the example script api_inpaint.py located in example/inpaint_example, but the returned result is still problematic. The preprocessed image returned is the same as the original image. The version of the web UI I used is v1.6.0-2-g4afaaf8a, and the controlnet version is 1.1.422.

txt2img inpaint result:
txt2img-0
txt2img-1

img2img inpaint result:
img2img-0
img2img-1

**Comment by AIGC404** [comment]
@huchenlei

**Comment by huchenlei** [comment]
I think your I paint result is correct. Somehow preprocessor image is not correctly shown though. 

The preprocessor result has alpha channel decreased in the inpaint area. I'm not sure what causes your result not showing anything as PNG should preserve alpha channel data.

**Comment by AIGC404** [comment]
When using inpaint_only+lama, although a preprocessor image is returned, it still differs from the one in the web UI. However, the generated results for image inpainting remain inconsistent.

**Comment by tiimgreen** [comment]
The problem here appears to be the mask. Your mask works, mine doesn't.

Can you explain how you made the mask image in this PR? @huchenlei

**Comment by huchenlei** [comment]
See following code in the example:
```python
def read_image(img_path: str) -> str:
    img = cv2.imread(img_path)
    _, bytes = cv2.imencode(".png", img)
    encoded_image = base64.b64encode(bytes).decode("utf-8")
    return encoded_image


input_image = read_image("1girl.png")
mask_image = read_image("mask.png")
```

**Comment by tiimgreen** [comment]
I mean the `mask.png` file, how did you generate that?

**Comment by tiimgreen** [comment]
I see that your image consists of 0s and 1s which creates the black/white.

But mine are on a scale of 0-255 so 0 and 1 are basically the same colour. If I try reverse engineer that it doesn't work, can you share how you created the mask image?

**Comment by huchenlei** [comment]
I did nothing special. I just export the image from photoshop as png.

If you can get np array of 0~255 value, you can just get the normalized np array by `img = img / 255`.
